driver/ssh/driver_test.go

```diff
@@ -1,65 +1,113 @@
+	"github.com/rudderlabs/compose-test/testcompose"
 )
 
-func TestMain(m *testing.M) {
-	privateKey, _ = ioutil.ReadFile("testdata/test_key")
-	m.Run()
-}
+func pingSSH(t *testing.T, dest string, config sshdriver.Config) error {
+	t.Helper()
 
-func TestSqlSSHPingsPostgresDB(t *testing.T) {
-	config := ssh.SSHConfig{
-		SshUser:    "root",
-		SshPort:    2222,
-		SshHost:    "0.0.0.0",
-		PrivateKey: privateKey,
+	singer, err := ssh.ParsePrivateKey(config.PrivateKey)
+	if err != nil {
+		return fmt.Errorf("test: parsing private key: %s", err.Error())
 	}
 
-	encoded, err := config.EncodeWithDSN("postgres://postgres:postgres@db_postgres:5432/postgres?sslmode=disable")
-	require.Nil(t, err)
+	client, err := ssh.Dial(
+		"tcp",
+		fmt.Sprintf("%s:%d", config.Host, config.Port),
+		&ssh.ClientConfig{
+			User:            config.User,
+			Auth:            []ssh.AuthMethod{ssh.PublicKeys(singer)},
+			HostKeyCallback: ssh.InsecureIgnoreHostKey(),
+			BannerCallback:  ssh.BannerDisplayStderr(),
+		},
+	)
+	if err != nil {
+		return fmt.Errorf("test: ssh dialing: %s", err.Error())
+	}
+	defer client.Close()
 
-	db, err := sql.Open("sql+ssh", encoded)
-	require.Nil(t, err)
+	t.Log(client.ServerVersion())
 
-	err = db.Ping()
-	require.Nil(t, err)
+	conn, err := client.Dial("tcp", dest)
+	if err != nil {
+		return fmt.Errorf("test: connection %q dialing: %s", dest, err.Error())
+	}
+	defer conn.Close()
 
-	err = db.Close()
-	require.Nil(t, err)
+	return nil
 }
 
-// Clickhouse
-func TestSqlSSHPingsClickhouseDB(t *testing.T) {
-	config := ssh.SSHConfig{
-		SshUser:    "root",
-		SshPort:    2222,
-		SshHost:    "0.0.0.0",
+func TestConnections(t *testing.T) {
+	t.Parallel()
+
+	privateKey, err := os.ReadFile("testdata/test_key")
+	require.Nil(t, err)
+
+	c := testcompose.New(t, "./testdata/docker-compose.yaml")
+
+	t.Cleanup(func() {
+		c.Stop(context.Background())
+	})
+	c.Start(context.Background())
+
+	config := sshdriver.Config{
+		User:       c.Env("openssh-server", "USER_NAME"),
+		Host:       "0.0.0.0",
+		Port:       c.Port("openssh-server", 2222),
 		PrivateKey: privateKey,
 	}
 
-	encoded, err := config.EncodeWithDSN("clickhouse://clickhouse:clickhouse@db_clickhouse:9000/db?skip_verify=false")
-	require.Nil(t, err)
+	t.Run("postgres", func(t *testing.T) {
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+		defer cancel()
 
-	db, err := sql.Open("sql+ssh", encoded)
-	require.Nil(t, err)
+		encoded, err := config.EncodeWithDSN("postgres://postgres:postgres@db_postgres:5432/postgres?sslmode=disable")
+		require.NoError(t, err)
 
-	err = db.Ping()
-	require.Nil(t, err)
+		db, err := sql.Open("sql+ssh", encoded)
+		require.NoError(t, err)
 
-	err = db.Close()
-	require.Nil(t, err)
+		err = db.PingContext(ctx)
+		require.NoError(t, err)
+
+		err = db.Close()
+		require.NoError(t, err)
+	})
+
+	t.Run("clickhouse", func(t *testing.T) {
+		// FIXME: hack to wait for clickhouse to be ready
+		require.Eventually(t, func() bool {
```

Comment:
Since `openssh-server` already depends on `db_postgres` and `db_clickhouse`. Do we still need this?

Reply:
This is something I still try to get my head around. Although I have setup depends_on and proper health checks, `clickhouse` connection will fail unless I sleep or retry a couple of times. To make matters even more strange, this doesn't happen for Postgres.